src/SFA.DAS.EAS.Portal.UnitTests/Portal/Application/Commands/Cohort/CohortApprovalRequestedCommandTests.cs

```diff
@@ -58,52 +61,26 @@ public TestContext()
                     .Setup(m => m.GetProviderCommitment(It.IsAny<long>(), It.IsAny<long>()))
                     .ReturnsAsync(TestCommitment);
 
-                Sut = new CohortApprovalRequestedCommandHandler(MockAccountsService.Object, MockProviderCommitmentsApi.Object, MockHashingService.Object, MockAccountHelperService.Object);
+                Fixture = new Fixture();
+                CohortApprovalRequestedByProvider = Fixture.Create<CohortApprovalRequestedByProvider>();
+                
+                CohortApprovalRequestedCommand = new CohortApprovalRequestedCommand(MockAccountDocumentService.Object, MockProviderCommitmentsApi.Object, MockHashingService.Object, MockLogger.Object);
             }
         }
 
         public class Handle : CohortApprovalRequestedCommandHandlerTests
         {
-            [Test]
-            public async Task WhenCalled_ThenTheAccountServiceIsCalledToRetrieveTheAccount()
-            {
-                // arrange
-                var testContext = new TestContext();
-                CohortApprovalRequestedCommand command = new CohortApprovalRequestedCommandBuilder();
-
-                // act
-                await testContext.Sut.Handle(command);
-
-                //assert
-                testContext.MockAccountHelperService.Verify(m => m.GetOrCreateAccount(command.AccountId, It.IsAny<CancellationToken>()), Times.Once);
-            }
-
-            [Test]
-            public async Task WhenCalled_ThenTheCommitmentsServiceIsCalledToRetrieveTheCommitmentForTheProvider()
-            {
-                // arrange
-                var testContext = new TestContext();
-                CohortApprovalRequestedCommand command = new CohortApprovalRequestedCommandBuilder();
-
-                // act
-                await testContext.Sut.Handle(command);
-
-                //assert
-                testContext.MockProviderCommitmentsApi.Verify(m => m.GetProviderCommitment(command.ProviderId, command.CommitmentId), Times.Once);
-            }
-
             [Test]
             public async Task WhenCalled_ThenTheAccountServiceIsCalledToSaveTheAccount()
             {
                 // arrange
                 var testContext = new TestContext();
-                CohortApprovalRequestedCommand command = new CohortApprovalRequestedCommandBuilder();
 
                 // act
-                await testContext.Sut.Handle(command);
+                await testContext.CohortApprovalRequestedCommand.Execute(testContext.CohortApprovalRequestedByProvider);
```

Comment:
What has this got to do with the story?  Please raise as tech debt for discussion if you think things need to change.

Reply:
Ideally yes, each tech debt item should have a separate PR.

However, in the initial creation phase of a project, when deadlines are so tight, it's useful to relax the overhead to get the system built in time, as you found useful in the [454 PR](https://github.com/SkillsFundingAgency/das-employerapprenticeshipsservice/pull/1632), which also contained many refactors 😉.

We should be in a position now though, going forward, to have separate PR's.